packages/url_launcher/url_launcher_ios/example/ios/RunnerTests/URLLauncherTests.swift

```diff
@@ -9,57 +9,42 @@ import XCTest
 
 final class URLLauncherTests: XCTestCase {
 
-  private func createPlugin() -> FLTURLLauncherPlugin {
+  private func createPlugin() -> URLLauncherPlugin {
     let launcher = FakeLauncher()
-    return FLTURLLauncherPlugin(launcher: launcher)
+    return URLLauncherPlugin(launcher: launcher)
   }
 
-  private func createPlugin(launcher: FakeLauncher) -> FLTURLLauncherPlugin {
-    FLTURLLauncherPlugin(launcher: launcher)
+  private func createPlugin(launcher: FakeLauncher) -> URLLauncherPlugin {
+    return URLLauncherPlugin(launcher: launcher)
   }
 
   func testCanLaunchSuccess() {
-    var error: FlutterError?
-    let result = createPlugin().canLaunchURL("good://url", error: &error)
+    let result = createPlugin().canLaunchUrl(url: "good://url")
 
-    XCTAssertNotNil(result)
-    XCTAssertTrue(result?.boolValue ?? false)
-    XCTAssertNil(error)
+    XCTAssertTrue(result)
   }
 
   func testCanLaunchFailure() {
-    var error: FlutterError?
-    let result = createPlugin().canLaunchURL("bad://url", error: &error)
+    let result = createPlugin().canLaunchUrl(url: "bad://url")
 
-    XCTAssertNotNil(result)
-    XCTAssertFalse(result?.boolValue ?? true)
+    XCTAssertFalse(result)
   }
 
   func testCanLaunchFailureWithInvalidURL() {
-    var error: FlutterError?
-    let result = createPlugin().canLaunchURL("urls can't have spaces", error: &error)
-
-    if (error == nil) {
-      // When linking against the iOS 17 SDK or later, NSURL uses a lenient parser, and won't
-      // fail to parse URLs, so the test must allow for either outcome.
-      XCTAssertNotNil(result)
-      XCTAssertFalse(result?.boolValue ?? true)
-      XCTAssertNil(error)
-    } else {
-      XCTAssertNil(result)
-      XCTAssertNotNil(error)
-      XCTAssertEqual(error?.code, "argument_error")
-      XCTAssertEqual(error?.message, "Unable to parse URL")
-      XCTAssertEqual(error?.details as? String, "Provided URL: urls can't have spaces")
-    }
+    let result = createPlugin().canLaunchUrl(url: "urls can't have spaces")
```

Comment:
why is this behavior changed?

Reply:
This seems like a merge mistake; changing this would re-break the tests for Xcode 15.

Reply:
I think I see the issue. The pigeon protocol replaced the objective-c error argument with the swift `throws` keyword. But Swift will allow for implementation not to have the throws keyword. So I made this method throw an exception. Now the test have do/catch blocks. This is a bit closer to the original behavior.